#! /bin/sh -e
## 62_new_list_bad_pending_requests.dpatch by richm@oldelvet.org.uk
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Only mail 'pending admin requests' if the database for the
## DP: particular list exists, so we do not send out '-1' requests.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
diff -urNad mailman-2.1.6~/Mailman/ListAdmin.py mailman-2.1.6/Mailman/ListAdmin.py
--- mailman-2.1.6~/Mailman/ListAdmin.py	2005-12-10 11:48:45.000000000 +0100
+++ mailman-2.1.6/Mailman/ListAdmin.py	2005-12-10 11:52:18.230056696 +0100
@@ -130,8 +130,11 @@
 
     def NumRequestsPending(self):
         self.__opendb()
-        # Subtract one for the version pseudo-entry
-        return len(self.__db) - 1
+        if self.__db.has_key('version'):
+            # Subtract one for the version pseudo-entry
+            return len(self.__db) - 1
+        else:
+            return len(self.__db)
 
     def __getmsgids(self, rtype):
         self.__opendb()
